web_widget_date_interval/static/src/js/web_widget_date_interval.js

```diff
@@ -0,0 +1,155 @@
+// -*- coding: utf-8 -*-
+// © 2017 Therp BV <http://therp.nl>
+// License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+/* eslint require-jsdoc:0 */
+
+openerp.web_widget_date_interval = function (instance) {
+    "use strict";
+    instance.web.form.widgets.add(
+        'date_interval',
+        'instance.web_widget_date_interval.FieldDateInterval');
+    instance.web_widget_date_interval.FieldDateInterval =
+    instance.web.form.AbstractField.extend(
+        instance.web.form.ReinitializeFieldMixin,
+        {
+            events: {
+                'click .weeknumber_iso .years button':
+                'weeknumber_iso_onchange_year',
+                'change .weeknumber_iso .weeks select':
+                'weeknumber_iso_onchange_week',
+                'click .weeknumber_iso .weeks button':
+                'weeknumber_iso_onchange_week',
+            },
+            default_options: {
+                type: 'weeknumber_iso',
+                years_before: 5,
+                years_after: 5,
+                weeks_before: 5,
+                weeks_after: 5,
+                hide_years: false,
+                week_type: 'select',
+            },
+            template: 'FieldDateInterval',
+            className: 'oe_form_date_interval',
+            init: function () {
+                this._super.apply(this, arguments);
+                this.options = _.extend(
+                    {}, this.default_options, this.options
+                );
+                if (!this.options.end_field) {
+                    throw new Error(
+                        'Pass an end field in the options dictionary');
```

Comment:
I modified the view from the front end and this was poping up all the time and I had to use postgres replace functions to fix the view. Maybe remove it and silently fail? The users should read `usage` on readme.